librosa/segment.py

```diff
@@ -488,16 +525,17 @@ def recurrence_matrix(
     for diag in range(-width + 1, width):
         rec.setdiag(0, diag)
 
-    # Retain only the top-k links per point
-    for i in range(t):
-        # Get the links from point i
-        links = rec[i].nonzero()[1]
+    if not full:
```

Comment:
If we're in full mode, does it make sense to hollow out the diagonal in line 525?

Reply:
I think you are right / it wouldn't make sense to hollow out the diagonal. That would mean `width` will also be ignored when in full mode. If the user wants to hollow out the diagonal it's pretty easy for them to do post-hoc as well, although this would have an effect on the bandwidth calculation...